libflux/flux-core/src/scanner/mod.rs

```diff
@@ -34,31 +35,44 @@ pub struct Scanner {
     checkpoint_last_newline: i32,
     token: TokenType,
     positions: HashMap<Position, u32, DefaultHasher>,
+
+    /// Comments
     pub comments: Vec<Comment>,
 }
 
+/// A position in source code.
 #[derive(Debug, PartialEq, Copy, Clone, Hash)]
 pub struct Position {
+    /// Line number.
     pub line: u32,
+    /// Column number.
     pub column: u32,
 }
 
 impl std::cmp::Eq for Position {}
 
+/// A token.
 #[derive(Debug, Display, PartialEq, Clone)]
 #[display(fmt = "{}", lit)]
 pub struct Token {
+    /// Type of token.
     pub tok: TokenType,
+    /// String representation of token.
     pub lit: String,
+    /// Offset
     pub start_offset: u32,
+    /// Offset
     pub end_offset: u32,
```

Comment:
What exactly are these offsets? Characters from the start of a file?

Reply:
Yes. Its the start and end of the current token within the context of the file.

Reply:
"Starting location of token" and "ending location of token" or something slightly better worded would probably be more accurate.

Reply:
Offset in character units, or tokens?

Reply:
Characters, I'm pretty sure. Not 100% sure if that includes whitespace or not.

Reply:
I think whitespace is left out. If that matters.

Reply:
Ah! Definitely characters. :) Always look at the tests, I guess.

https://github.com/influxdata/flux/blob/8d902fdf517445a31af22065541a30584bcc49ef/libflux/flux-core/src/scanner/tests.rs#L4-L18